File: domain/src/main/java/org/openmedstack/eventstore/Commit.java
package org.openmedstack.eventstore;

import java.time.Instant;
import java.util.HashMap;
import java.util.List;
import java.util.UUID;

public interface Commit {
    String getBucketId();

    String getStreamId();

    int getStreamRevision();

    UUID getCommitId();

    int getCommitSequence();

    Instant getCommitStamp();

    HashMap<String, Object> getHeaders();

    List<EventMessage> getEvents();

    long getCheckpointToken();
}

File: domain/src/main/java/org/openmedstack/eventstore/Commit.kt
package org.openmedstack.eventstore

import java.time.Instant
import java.util.*

interface Commit {
    val bucketId: String?
    val streamId: String
    val streamRevision: Int
    val commitId: UUID
    val commitSequence: Int
    val commitStamp: Instant
    val headers: HashMap<String, Any>
    val events: MutableList<EventMessage>
    val checkpointToken: Long
}
